app/src/main/java/com/github/se/icebreakrr/model/ai/AiRepositoryChatGPT.kt

```diff
@@ -0,0 +1,139 @@
+package com.github.se.icebreakrr.model.ai
+
+import java.io.IOException
+import kotlin.coroutines.suspendCoroutine
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import org.json.JSONArray
+import org.json.JSONObject
+
+// Data class for encapsulating request parameters
+data class AiRequest(
+    val systemPrompt: String,
+    val userPrompt: String,
+    val model: String = "gpt-4o-mini", // Default model
+    val temperature: Double = 0.7,
+    val maxTokens: Int = 50
+)
+
+class AiRepositoryChatGPT(
+    private val apiKey: String,
+    private val baseUrl: String = "https://api.openai.com/v1/chat/completions"
+) : AiRepository {
+
+  private val client = OkHttpClient()
+
+  /**
+   * Generates a response from the AI model based on the given request parameters.
+   *
+   * @param request The encapsulated request parameters.
+   * @return The response text from the AI model.
+   * @throws IOException if an error occurs during the API call.
+   */
+  override suspend fun generateResponse(request: AiRequest): String =
+      withContext(Dispatchers.IO) {
+        // Validate input
+        validateRequest(request)
+        try {
+          // Construct the request body
+          val requestBody: String =
```

Comment:
For modularity and simplicity maybe could move this out of the body of generateResponse